src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalDetainedEditCaseListingLetterTemplateTest.java

```diff
@@ -0,0 +1,109 @@
+        when(asylumCase.read(APPELLANT_GIVEN_NAMES, String.class)).thenReturn(Optional.of(appellantGivenNames));
+        when(asylumCase.read(APPELLANT_FAMILY_NAME, String.class)).thenReturn(Optional.of(appellantFamilyName));
+        when(asylumCase.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)).thenReturn(Optional.of(HearingCentre.MANCHESTER));
+        when(asylumCase.read(LIST_CASE_HEARING_DATE, String.class)).thenReturn(Optional.of(listCaseHearingDate));
+        when(asylumCaseBefore.read(LIST_CASE_HEARING_DATE, String.class)).thenReturn(Optional.of(hearingDateBefore));
+        when(asylumCaseBefore.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)).thenReturn(Optional.of(HearingCentre.TAYLOR_HOUSE));
+        when(stringProvider.get("hearingCentreName", "taylorHouse")).thenReturn(Optional.of(expectedFormattedTaylorHouseHearingCentreName));
+        when(stringProvider.get("hearingCentreAddress", "manchester")).thenReturn(Optional.of(manchesterHearingCentreAddress));
+        when(customerServicesProvider.getInternalCustomerServicesTelephone(asylumCase)).thenReturn(customerServicesTelephone);
+        when(customerServicesProvider.getInternalCustomerServicesEmail(asylumCase)).thenReturn(customerServicesEmail);
+    }
+
+    @Test
+    void should_map_case_data_to_template_field_values() {
+        dataSetUp();
+
+        Map<String, Object> templateFieldValues = internalDetainedEditCaseListingLetterTemplate.mapFieldValues(caseDetails, caseDetailsBefore);
+
+        assertEquals(14, templateFieldValues.size());
+        assertEquals(appealReferenceNumber, templateFieldValues.get("appealReferenceNumber"));
+        assertEquals(homeOfficeReferenceNumber, templateFieldValues.get("homeOfficeReferenceNumber"));
+        assertEquals(appellantGivenNames, templateFieldValues.get("appellantGivenNames"));
+        assertEquals(appellantFamilyName, templateFieldValues.get("appellantFamilyName"));
+        assertEquals(formatDateForNotificationAttachmentDocument(now), templateFieldValues.get("dateLetterSent"));
+        assertEquals(formattedListCaseHearingDate, templateFieldValues.get("hearingDate"));
+        assertEquals(formattedListCaseHearingTime, templateFieldValues.get("hearingTime"));
+        assertEquals(formattedListCaseHearingDateBefore, templateFieldValues.get("oldHearingDate"));
+        assertEquals(formattedListCaseHearingTimeBefore, templateFieldValues.get("oldHearingTime"));
+        assertEquals(expectedFormattedTaylorHouseHearingCentreName, templateFieldValues.get("oldHearingCentre"));
+        assertEquals(formattedManchesterHearingCentreAddress, templateFieldValues.get("hearingCentreAddress"));
+        assertEquals(customerServicesTelephone, templateFieldValues.get("customerServicesTelephone"));
+        assertEquals(customerServicesEmail, templateFieldValues.get("customerServicesEmail"));
+    }
+}
```

Comment:
would be good to have tests for situations where previous case data/current case data is missing and assert an exception is thrown.

Reply:
Good call, added the tests